src/App.js

```diff
@@ -25,34 +26,106 @@ const generateSquares = () => {
   return squares;
 }
 
+// Tracks movements in game so far.
+let trackBoard = [
+  "", "", "",
+  "", "", "",
+  "", "", "",
+];
```

Comment:
A couple of things here:

1. This is basically a duplicate of `squares` so now you have two "sources of truth" for this information and they could get out of sync.
2. You shouldn't use a global `let` to keep track of state using React.  We want to use the `useState` hook.  This works okay here because you only ever make one `App` component but would get you in trouble if this were a component like `Square` that you want to make several of.

If you would like to just get a flat array out of `squares` to work with you can do the following:
```javascript
squares.flat().map(s => s.value)
```